T_PHY_002 modified to address explicitely fault injection attacks

ronan-lashermes opened the pull request:
As discussed in #49, a new formulation for T_PHY_002 addressing explicitly fault injection.

yloisel commented on specification/src/chapter2.adoc:
```diff
@@ -545,19 +545,17 @@ a| * Implement robust power management and radiation control
 
 | T_PHY_002
-| Physical memory manipulation
+| Fault Injection
```
I would prefer we keep both: Physical Memory Manipulation and Fault Injection. Both are meaningful

rsahita replied:
agree - add a T_PHY_003

andrewdellow replied:
IMO FI is a separate threat from most of the physical memory manipulation as described (probing, interposers etc) whereas FI is modifying circuit state  (EMFI, laser, voltage/clock/reset glitch etc ). Row hammer is FI specific to memory, and isn't really physical manipulation, so perhaps we  add a T_PHY_03 - fault Injection attack, and move row hammer under that ?

rsahita commented on specification/src/chapter2.adoc:
```diff
@@ -545,19 +545,17 @@ a| * Implement robust power management and radiation control
+* Using NVDIMM, interposers, or physical probing to read, record, or replay physical memory  
+* Fault injection attacks (glitching, laser, electromagnetic, ...) to extract hardware-provisioned assets  
+a| * Implement robust memory error detection, cryptographic memory protection, or physical tamper resistance  
+* Use supervisor domain ID, privilege level, or MTT attributes to derive memory encryption contexts at the domain or workload granularity  
```
```suggestion
* Use supervisor domain ID, privilege level, and/or MPT permissions to derive memory access control / encryption contexts at the domain or workload granularity  
```

andrewdellow replied:
FI isn't only used to extract hardware provisioned state but also to modify circuit behaviour, bypassing isolation or confidentiality protections

rsahita reviewed (changes requested):
lgtm with changes

andrewdellow reviewed (changes requested):
I prefer we leave T_PHY_02, remove rowhammer reference, and add new T_PHY_03 Fault Injection which includes row hammer.

ronan-lashermes commented:
Ok, I split the Fault Injection into a new T_PHY_005 entry.
I removed the phrase `* Physical attacks on hardware shielded locations to extract hardware provisioned assets` from T_PHY_002. Too ambiguous in my opinion, but I don't know how to properly replace it.

andrewdellow commented on specification/src/chapter2.adoc:
```diff
@@ -579,6 +583,18 @@ debug authorizations)
+* Provide a level of tamper resistance, e.g., through RoT attestation, redundancy during execution, etc.
+* Enforce proper access control for the DVFS configuration.
+* Employ lockstep processors for security-critical devices.
+
```
suggest the use of sensors ?

andrewdellow replied:
```suggestion
* Employ physical sensors to detect attack
```

andrewdellow replied:
IGTM

rsahita commented on specification/src/chapter2.adoc:
```diff
@@ -544,20 +544,22 @@ Physical or remote
 a| * Implement robust memory error detection, cryptographic memory protection,
-or physical tamper resistance
+or physical tamper resistance.
 * Supervisor domain ID, privilege level, or MTT attributes, may be used to
```
```suggestion
* Supervisor domain ID, privilege level, or MPT attributes, may be used to
```
